src/ShaderUtils.cpp

```diff
@@ -56,18 +56,42 @@ void logErrorShader(GLenum _shaderType, const std::string & _strShader)
 	}
 }
 
+const char* addGLSLVersion(const char* shaderString)
+{
+	OGLVideo & ogl = video();
+	OGLRender & render = ogl.getRender();
+	std::string GLversion = std::to_string(render.majorVersion) + std::to_string(render.minorVersion);
+#if defined(GLES2)
+	finalString = "#version 100 \n";
+	if (config.generalEmulation.enableLOD > 0) {
+		finalString += "#extension GL_EXT_shader_texture_lod : enable \n" \
+			"#extension GL_OES_standard_derivatives : enable \n";
+	}
+#elif defined(GLESX)
+	finalString = "#version "+ GLversion + "0 es \n";
+#else
+	finalString = "#version "+ GLversion + "0 core \n";
+	if (video().getRender().isImageTexturesSupported() && stoi(GLversion) < 42)
+		finalString += "#extension GL_ARB_shader_image_load_store : enable \n";
+#endif
+	finalString.append(shaderString);
+	return finalString.c_str();
+}
+
```

Comment:
I don't like that the strings calculated each time addGLSLVersion called.

I suggest:

+const std::string &  getHeader()
 +{
       static std::string strHeader;
       if (!strHeader().empty())
          return strHeader;


 +	OGLVideo & ogl = video();
 +	OGLRender & render = ogl.getRender();
 +	std::string GLversion = std::to_string(render.majorVersion) + std::to_string(render.minorVersion);
 +#if defined(GLES2)
 +	strHeader= "#version 100 \n";
 +	if (config.generalEmulation.enableLOD > 0) {
 +		strHeader+= "#extension GL_EXT_shader_texture_lod : enable \n" \
 +			"#extension GL_OES_standard_derivatives : enable \n";
 +	}
 +#elif defined(GLESX)
 +	strHeader= "#version "+ GLversion + "0 es \n";
 +#else
 +	strHeader= "#version "+ GLversion + "0 core \n";
 +	if (video().getRender().isImageTexturesSupported() && stoi(GLversion) < 42)
 +		strHeader+= "#extension GL_ARB_shader_image_load_store : enable \n";
 +#endif
 +	return strHeader;
 +}

const char* addGLSLVersion(const char* shaderString)
 +{
       static std::string strResult;
        strResult = getHeader();
 +	strResult .append(shaderString);
 +	return strResult .c_str();
 +}